packages/core/src/vector/in-memory-vector.ts

```diff
@@ -0,0 +1,110 @@
+import { sum } from 'radash';
+import type {
+  CreateIndexParams,
+  UpsertVectorParams,
+  QueryVectorParams,
+  IndexStats,
+  ParamsToArgs,
+  QueryResult,
+  CreateIndexArgs,
+  UpsertVectorArgs,
+  QueryVectorArgs,
+} from './types';
+import { MastraVector } from './vector';
+type VectorDoc = {
+  id: string;
+  vector: number[];
+  metadata: Record<string, unknown>;
+};
+
+export class InMemoryVector extends MastraVector {
```

Comment:
Why can we not use the same idea for this as we do for regular storage and use libsql?